src/lib/consensus/proof_of_stake.ml

```diff
@@ -331,7 +329,8 @@ module Data = struct
       type t =
         { mutable last_epoch_snapshot: Snapshot.t
         ; mutable curr_epoch_snapshot: Snapshot.t
-        ; mutable last_checked_slot_and_epoch: Epoch.t * Epoch.Slot.t
+        ; last_checked_slot_and_epoch:
+            (Epoch.t * Epoch.Slot.t) Public_key.Compressed.Table.t
```

Comment:
Doesn't it make more sense to track this in the delegator table? Then when you update keypairs, you would take your old table, remove the keys that are no longer there, and process the new keys with the epoch/slot set to current time.

Reply:
Hmm there isn't really one delegator table. There are ones for each snapshot -- so it's not clear exactly where this should be stored. I can (and will) derive the proposer_public_keys field from the keys of this table however (as that will deal with filtering out the ones that aren't in the ledger also).